ipfx/bin/run_chirp_fv_extraction.py

```diff
@@ -0,0 +1,176 @@
+import numpy as np
+import argschema as ags
+import lims_utils
+import ipfx.chirp as chirp
+from ipfx.aibs_data_set import AibsDataSet
+import logging
+import traceback
+from multiprocessing import Pool
+from functools import partial
+import h5py
+import os
+import json
+import allensdk.core.json_utilities as ju
+from ipfx.stimulus import StimulusOntology
+from ipfx.bin.run_feature_vector_extraction import lims_nwb_information, sdk_nwb_information
+
+class CollectChirpFeatureVectorParameters(ags.ArgSchema):
+    input_file = ags.fields.InputFile(
+        description="input text file with specimen numbers of each line",
+        default=None, allow_none=True)
+    include_failed_cells = ags.fields.Boolean(
+        description="whether to include failed cells",
+        default=False)
+    run_parallel = ags.fields.Boolean(
+        description="whether to use multiprocessing",
+        default=True)
+    output_dir = ags.fields.OutputFile(
+        description="output directory", default=None)
+    output_code = ags.fields.String(
+        description="output code for naming files", default=None)
+    chirp_stimulus_codes = ags.fields.List(ags.fields.String,
+        description="stimulus code for chirps",
+        default=[
+            "C2CHIRP180503",
+            "C2CHIRP171129",
+            "C2CHIRP171103",
+        ],
+        cli_as_single_argument=True)
+    data_source = ags.fields.String(
+        description="Source of NWB files ('sdk' or 'lims')",
+        default="sdk",
+        validate=lambda x: x in ["sdk", "lims"]
+        )
+
+
+def data_for_specimen_id(specimen_id, data_source, chirp_stimulus_codes):
+    logging.debug("specimen_id: {}".format(specimen_id))
+
+    # Manual edit ontology to identify chirp sweeps
+    ontology_data = ju.read(StimulusOntology.DEFAULT_STIMULUS_ONTOLOGY_FILE)
+    mask = []
+    for od in ontology_data:
+        mask_val = True
+        for tagset in od:
+            for c in chirp_stimulus_codes:
+                if c in tagset and "code" in tagset:
+                    mask_val = False
+                    break
+        mask.append(mask_val)
+    ontology_data = [od for od, m in zip(ontology_data, mask) if m is True]
+    ontology_data.append([
+        ["code"] + chirp_stimulus_codes,
+        [
+          "name",
+          "Chirp",
+        ],
+        [
+          "core",
```

Comment:
This seems like two questions. First, the chirp stimuli are coded as "unknown" in the default stimulus ontology, so I'm not sure how else to tag them appropriately. There is no "Chirp" stimulus type listed in LIMS (and I don't have the ability to add it there).

For the second question, it's a good idea to edit the ontology once rather than for every specimen, so I've moved the code into its own function and passed the updated ontology instead of a list of stimulus codes, as you suggested.